### PR TITLE
[MCKIN-5641] OAuth2 API to call XBlock handlers

### PR DESCRIPTION
**Pull request by cgopalan**
[MCKIN-5641] OAuth2 API to call XBlock handlers

**JIRA tickets**: [OC-2996](https://tasks.opencraft.com/browse/OC-2996)

**Reviewers**
- [ ] @mtyaka

### CODE REVIEW

File: lms/djangoapps/courseware/module_render.py [hunk 1]
```diff
@@ -1013,7 +1005,10 @@ def _invoke_xblock_handler(request, course_id, usage_id, handler, suffix, course
     """
 
     # Check submitted files
-    files = request.FILES or {}
+    # files = request.FILES or {}
```
Comment (mtyaka) [inline lms/djangoapps/courseware/module_render.py]: Please remove commented-out code.

File: lms/djangoapps/courseware/tests/test_module_render.py [hunk 2]
```diff
@@ -268,7 +269,7 @@ def test_anonymous_handle_xblock_callback(self):
         )
         response = self.client.post(dispatch_url, {'position': 2})
         self.assertEquals(403, response.status_code)
-        self.assertEquals('Unauthenticated', response.content)
+        #self.assertEquals('Unauthenticated', response.content)
```
Comment (mtyaka) [inline lms/djangoapps/courseware/tests/test_module_render.py]: Please remove commented-out code.

File: lms/djangoapps/courseware/module_render.py [hunk 3]
```diff
@@ -907,36 +911,24 @@ def handle_xblock_callback_noauth(request, course_id, usage_id, handler, suffix=
...
 
-    with modulestore().bulk_operations(course_key):
-        try:
-            course = modulestore().get_course(course_key)
-        except ItemNotFoundError:
-            raise Http404("invalid location")
 
-        return _invoke_xblock_handler(request, course_id, usage_id, handler, suffix, course=course)
+    def post(self, request, course_id, usage_id, handler, suffix=None):
+        return _get_course_and_invoke_handler(request, course_id, usage_id, handler, suffix)
```
Comment (mtyaka) [inline lms/djangoapps/courseware/module_render.py]: Can we also add `def put(...)`, `def delete(...)`, and `def patch(...)`? They should all just proxy to  `_get_course_and_invoke_handler`.

**Comment by cgopalan** [comment]
@mtyaka Review comments addressed and commits squashed.

**Comment by mtyaka** [comment]
Thanks @cgopalan!

**Comment by bradenmacdonald** [comment]
@mtyaka Please remember trigger a CircleCI build on this repo by pushing the branch to the `edx-solutions` fork. In this case the build wasn't passing due to an unrelated issue.

**Comment by mtyaka** [comment]
@bradenmacdonald 

> @mtyaka Please remember trigger a CircleCI build on this repo by pushing the branch to the edx-solutions fork. In this case the build wasn't passing due to an unrelated issue.

Oh, I didn't know about that. Thanks for the info, will remember for next time.

**Comment by bradenmacdonald** [comment]
@cgopalan (CC @mtyaka) Yeah actually this introduced some PEP8 issues in `test_module_render.py`, causing the builds to fail. Can you please fix and submit a new PR ASAP?

**Comment by mtyaka** [comment]
@bradenmacdonald Sorry about that :/ I opened https://github.com/edx-solutions/edx-platform/pull/859 to address that.

**Pull request merged**